### DISCUSSION TITLE
multiple definition of `tud_cdc_n_connected'

### DISCUSSION THREAD

**Discussion by arilink-tech** [hathach/tinyusb#698] [General]
When I select CFG_TUD_HID and CFG_TUD_CDC in tusb_config.h->class and add desc to usb_descriptors.c, there are 35 redefinition errors when compiling. It is about dfu and cdc. I removed the dfu source file and the error disappeared but I still need it later.  
What caused this problem?
![image](https://user-images.githubusercontent.com/30901123/109932789-9c912b00-7d05-11eb-866a-27f2800aa366.png)

**Comment by chegewara** [comment]
It looks like you are having issue with folders, maybe 2 versions of tinyusb installed. Folder `common` should be inside `src`.

**Reply by arilink-tech** [reply]
My current project directory is like this, there is only 1 version

**Reply by arilink-tech** [reply]
![image](https://user-images.githubusercontent.com/30901123/109958424-2d293480-7d21-11eb-859c-48273a9e03fc.png)
